tss-esapi/src/abstraction/web_authn.rs

```diff
@@ -0,0 +1,181 @@
+use crate::{
+    constants::SessionType,
+    handles::{KeyHandle, SessionHandle},
+    interface_types::algorithm::HashingAlgorithm,
+    structures::{
+        Attest, HashScheme, PcrSelectionList, Public, Signature, SignatureScheme,
+        SymmetricDefinition,
+    },
+    traits::Marshall,
+    Context, Error, Result,
+    WrapperErrorKind::InternalError,
+};
+use ciborium::{cbor, ser::into_writer};
+use std::convert::TryInto;
+
+#[derive(Debug)]
+pub struct TpmStatement {
+    attestation: Attest,
+    signature: Signature,
+    public_area: Public,
+    algorithm: SignatureScheme,
+    x509_chain: Vec<Vec<u8>>,
+}
+
+impl TpmStatement {
+    pub fn new(
+        context: &mut Context,
+        object: KeyHandle,
+        key: KeyHandle,
+        nonce: Vec<u8>,
+    ) -> Result<TpmStatement> {
+        let (public_area, _, _) = context.read_public(object)?;
+        let session_1 = context.start_auth_session(
+            None,
+            None,
+            None,
+            SessionType::Hmac,
+            SymmetricDefinition::AES_128_CFB,
+            HashingAlgorithm::Sha256,
+        )?;
+        let session_2 = context.start_auth_session(
+            None,
+            None,
+            None,
+            SessionType::Hmac,
+            SymmetricDefinition::AES_128_CFB,
+            HashingAlgorithm::Sha256,
+        )?;
+        let signing_scheme = SignatureScheme::RsaSsa {
+            hash_scheme: HashScheme::new(HashingAlgorithm::Sha256),
+        };
```

Comment:
We're gonna need to make this dependent on the parameters of `key`, but not necessarily now. 

If you want to have a go at this, you can try to read the [public area](https://docs.rs/tss-esapi/7.1.0/tss_esapi/structures/enum.Public.html) of `key` (as you did for `object`), then convert the [`RsaScheme`](https://docs.rs/tss-esapi/7.1.0/tss_esapi/structures/struct.PublicRsaParameters.html#method.rsa_scheme) or [`EccScheme`](https://docs.rs/tss-esapi/7.1.0/tss_esapi/structures/struct.PublicEccParameters.html#method.ecc_scheme) (from the `parameters` field in the public area) to [`SignatureScheme`](https://docs.rs/tss-esapi/7.1.0/tss_esapi/structures/enum.SignatureScheme.html). You have to do this manually, there's no `From`/`TryFrom`...

Otherwise, just add a TODO comment

Reply:
(and same applies for the `TpmPlatStatement` constructor too)